kernel/CMakeLists.txt

```diff
@@ -30,6 +30,12 @@ zephyr_syscall_header_ifdef(
   ${ZEPHYR_BASE}/include/zephyr/kernel/mm/demand_paging.h
 )
 
+zephyr_linker_sources_ifdef(
+  CONFIG_DEVICE_DT_METADATA
+  SECTIONS
+  ${ZEPHYR_BASE}/include/zephyr/linker/device-dt-metadata-section.ld
+)
```

Comment:
I've had issues with using the `SECTIONS` location, as this is added in the `RAMABLE_REGION` (this is an assumption, I'm no linker expert).

I've created PR #71911 to allow putting sections after the `common-rom` sections (where device structs are located)

Reply:
Could you explain what the issue is? We're using `SECTIONS` all over the place, so I'm guessing you mean there are certain situations where it's not working as desired, and this is one of them. If so, can you clarify?

Reply:
So either you will need to add the section to, for example `include/zephyr/linker/common-rom/common-rom-kernel-devices.ld`, or use something like my proposed `ROM_SECTIONS`, I think.

Reply:
Cool PR! OK, I'll check twister results for a similar error and see if your #71911 helps in case I  see them too. Thanks for explaining.

Reply:
Hmm, interesting. If nobody has a better idea, I guess I can try to rework this to make a pointer to device_dt_metadata from struct device, like we do with other state.

Reply:
@pdgendt you know, it was actually cleaner to do it that way and not use SECTIONS at all. Now updated to just have all the DT metadata be reachable from struct device itself, so we can iterate over devices to iterate over their metadata.